src/stores/TagOrderStore.js

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2017 Vector Creations Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import {Store} from 'flux/utils';
+import dis from '../dispatcher';
+
+const INITIAL_STATE = {
+    orderedTags: null,
+    allTags: null,
+};
+
+/**
+ * A class for storing application state for ordering tags in the TagPanel.
+ */
+class TagOrderStore extends Store {
+    constructor() {
+        super(dis);
+
+        // Initialise state
+        this._state = INITIAL_STATE;
+    }
+
+    _setState(newState) {
+        this._state = Object.assign(this._state, newState);
+        this.__emitChange();
+    }
+
+    __onDispatch(payload) {
+        switch (payload.action) {
+            // Get ordering from account data
+            case 'MatrixActions.accountData': {
+                if (payload.event_type !== 'im.vector.web.tag_ordering') break;
+                this._setState({
+                    orderedTags: payload.event_content ? payload.event_content.tags : null,
+                });
```

Comment:
Do we fetch the tag order initially anywhere? How does it initialise `orderedTags`?

Reply:
This should be done on `sync_state` or whatever the equivalent ends up being.